library/Vanilla/Utility/UrlUtils.php

```diff
@@ -35,4 +38,46 @@ public static function domainAsAscii(string $url): ?string {
         $buildUrl = http_build_url($parsedLink);
         return $buildUrl;
     }
+
+    /**
+     * Generate a new URI by replacing querystring elements from an existing URI.
+     *
+     * @param UriInterface $uri The base URI.
+     * @param array $replace The querystring replacement.
+     * @return UriInterface
+     */
+    public static function replaceQuery(UriInterface $uri, array $replace): UriInterface {
+        parse_str($uri->getQuery(), $query);
+        $query = array_replace($query, $replace);
+        $result = $uri->withQuery(http_build_query($query));
+        return $result;
```

Comment:
Do we still need the `http_build_query` and `parse_str` here?

Looks likes we could do this:

```
return UriModifier::mergeQuery($uri, QueryString::build($replace));
```

Reply:
I guess so, but I don't know if it's enough reason to change the implementation.

Reply:
For what it's worth, I don't think this is possible without requiring the [league/uri-components](https://packagist.org/packages/league/uri-components) package. We're only introducing [league/uri](https://packagist.org/packages/league/uri) here.